src/System.Net.Http/src/System/Net/Http/HttpRuleParser.cs

```diff
@@ -433,7 +433,7 @@ internal static HttpParseResult GetQuotedPairLength(string input, int startIndex
                     }
                 }
 
-                if (input[current] == closeChar)
+                if (current < input.Length &&  input[current] == closeChar)
```

Comment:
Why is this code changing here?  It seems like it will affect more than just the 'User-Agent' header.

Your PR description doesn't describe anything about this fix other than it fixes a known issue.   It's best if the PR description was more descriptive about why this code is changing etc.

And since this code is changing and affects more headers, more tests will be required to validate a change like this.

Reply:
I agree.

It looks like the root of the problem here is handling nested expressions. And it looks like the problem may be deeper than just this particular case, because (if I'm understanding correctly) we will ignore the character after the end of the nested expression unless it's the closeChar, which seems like the wrong thing to do if it's the start of another nested expression (i.e. openChar) or the start of a quoted string.

Reply:
Ok I'll work on it ASAP, thank's for your time!